src/core/qfieldutils.cpp

```diff
@@ -0,0 +1,36 @@
+#include "qfieldutils.h"
+
+#include <qgslinestring.h>
+#include <qgspolygon.h>
+#include <qgsvectorlayer.h>
+
+#include "rubberbandmodel.h"
+
+QFieldUtils::QFieldUtils(QObject *parent) : QObject(parent)
+{
+
+}
+
+QgsFeature QFieldUtils::initFeature(QgsVectorLayer *layer, QgsGeometry geometry)
+{
+  QgsFeature f(layer->fields());
+  f.setGeometry(geometry);
+  return f;
+}
+
+QgsGeometry QFieldUtils::polygonFromRubberband(RubberbandModel *rubberBandModel, const QgsCoordinateReferenceSystem &crs)
```

Comment:
Can we move this to geometryutils and featureutils etc?
I don't like this idea of a dumping ground

Reply:
done, I moved them to a subrepo